includes/Core/REST_API/REST_Routes.php

```diff
@@ -93,6 +93,33 @@ function( $server ) {
 				$this->register_routes( $server );
 			}
 		);
+
+		add_filter(
+			'do_parse_request',
+			function( $do_parse_request, $wp ) {
+				add_filter(
+					'query_vars',
+					function( $vars ) use ( $wp ) {
+						// Unsets standard public query vars to escape conflicts between WorPress core
+						// and Google Site Kit APIs which happen when WordPress incorrectly parses request
+						// arguments.
+						$namespace = rest_get_url_prefix() . '/' . self::REST_ROOT;
+						if ( substr( $wp->request, 0, strlen( $namespace ) ) === $namespace ) {
```

Comment:
This won't work for sites that are using plain permalinks as `$wp->request` is empty in this case. We can use this condition for cases where the site is using pretty permalinks, but for plain I'm not sure if there is a better option at this point in the request, and we may need to resort to a checking the `REQUEST_URI` but that's getting pretty hacky. Is there a way that you know of that we can reliably use for this case?

Reply:
We can fallback to `REQUEST_URI` if `$wp->request` is empty. Does it sound good to you? @felixarntz are you good with it?

Reply:
@eugene-manuilov I think that's reasonable. Let's add a test for this if possible, for both pretty and ugly permalinks (relying on `rest_route` query parameter).

There is quite some complexity in this though to ensure we cover all instances across different setups, e.g. we'd also need to account for `$wp_rewrite->index` etc. Although this wouldn't exactly be "precise", I'm starting to think we should just see if the request path or query parameter `rest_route` contains `/google-site-kit/v1`?

Reply:
@eugene-manuilov this seems to be the only thing left to address here.

I think what we need to update this as follows:
- If `$wp->did_permalink`
    - it is a Site Kit REST request if `$wp->request` _contains_ `wp-json/google-site-kit/v1` (not necessarily starts with, so that it covers the `$wp_rewrite->index` case)
- If not If `$wp->did_permalink`
   - it is a Site Kit REST request if `$this->context->input->filter( INPUT_GET, 'rest_route' )` starts with `google-site-kit/v1`  
    The `rest_route` query param takes precedence in this case, even if other parameters are present (which shouldn't be the case) but WP only checks for this query var, so this is enough for us too.

Reply:
@aaemnnosttv updated it slightly different. Decided not to rely on `$wp->did_permalink` because it might be reset back to `false` in a certain situation.